### PR TITLE
Don't show circular field when creating new item

### LINKED ISSUE #15053: Can't insert child item in sidebar
### Describe the Bug

After #14834 a side affect was that even though it was disabled it could still be required, thereby hindering creating a new item.

### To Reproduce

Create many to one relationship. In the one relationship try and create a new "many" item.

### Errors Shown

Value is required (even though it is disabled)

### What version of Directus are you using?

9.15.1

### What version of Node.js are you using?

16.4.0

### What database are you using?

mysql 8

### What browser are you using?

chromium

### How are you deploying Directus?

Locally & GCP

### PR DESCRIPTION
**Pull request by u12206050**
## Description

Just a refactor fix of the issue.

Turns out the issue is only on create, and since it will anyways be empty, we don't need to show the field during creation.

Fixes #15053

## Type of Change

- [ ] Bugfix
- [ ] Improvement
- [ ] New Feature
- [x] Refactor / codestyle updates
- [ ] Other, please describe:

### CODE REVIEW

**Comment by u12206050** [comment]
@rijkvanzanten Could you do a hot fix release on this one? It is stopping all our users from creating new items since the field is required.

**Review by br41nslug — CHANGES_REQUESTED** [review]
When testing this branch im still getting the "missing required field error" when i tick the `required` toggle. 

image

I believe this is due to the change to a `filter` function instead of a `map` without moving the `set(field, 'meta.readonly', true);` outside of the `filter` loop.

File: app/src/views/private/components/drawer-item.vue [hunk 1]
```diff
@@ -178,27 +178,21 @@ export default defineComponent({
...
 		const fields = computed(() => {
 			if (props.circularField) {
-				return fieldsWithPermissions.value.map((field: Field) => {
+				return fieldsWithPermissions.value.filter((field: Field) => {
 					if (field.field === props.circularField) {
+						if (props.primaryKey === '+') {
+							return false;
+						}
+
 						set(field, 'meta.readonly', true);
```
[the remark this answers is not in the document]
Comment (u12206050) [reply app/src/views/private/components/drawer-item.vue:187]: map and filter both returned the same references ie. field.
Map simply modified the field before returning it, and filter is also modifying it, but it is also filtering out the one field that is the circularField when creating, else during update it sets the field to readonly

**Comment by u12206050** [comment]
> When testing this branch im still getting the "missing required field error" when i tick the `required` toggle.
> 
> image
> 
> I believe this is due to the change to a `filter` function instead of a `map` without moving the `set(field, 'meta.readonly', true);` outside of the `filter` loop.

The error you are getting regarding the ID field, might be due to another issue I believe. Is the ID there an auto-increment ID field?

File: app/src/views/private/components/drawer-item.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (br41nslug) [reply app/src/views/private/components/drawer-item.vue:186]: It does work damn, gotta love JS 🤔 It does feel semantically wrong but was definitely more of a "how i think it should work" comment 😇

**Comment by br41nslug** [comment]
Can it not be an auto increment? i didnt get that from the linked ticket 😬

**Comment by u12206050** [comment]
> 

It probably can, that was just a question I asked, in regards to why you got that error. Not really anything more to do with this PR.

**Comment by Nitwel** [comment]
This PR will be closed as the underlying issue should be resolved by #15386.

**Pull request closed without merging**